File: 2023/day_01/src/part1.rs
fn main() {
    // read input.in
    let input = std::fs::read_to_string("src/input.in").unwrap();

    // split input into lines
    let lines = input.lines();

    let mut total = 0;

    // for each line
    for line in lines {
        let first = line.chars().find(|c| c.is_digit(10)).unwrap_or('0');
        let last = line.chars().rev().find(|c| c.is_digit(10)).unwrap_or('0');

        total += format!("{}{}", first, last).parse::<i32>().unwrap();
    }

    // print total
    println!("Total: {}", total);
}

File: 2016/day_20/part1.py
# open and read the file to a string
with open("input.in", "r") as file:
    lines = file.read().split("\n")

def lowest_unlocked_ip(range_list):
    range_list.sort()
    lowest = 0
    for start, end in range_list:
        if lowest < start:
            return lowest
        elif lowest <= end:
            lowest = end + 1
    return lowest

ranges = [list(map(int, line.split("-"))) for line in lines]
print(lowest_unlocked_ip(ranges))
